server/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPServlet.java

```diff
@@ -78,9 +78,9 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
                     .collect(Collectors.toList());
 
             resp.setContentType(JSON_CONTENT_TYPE);
-            objectMapper.writeValue(resp.getOutputStream(), responses);
+            sendResponses(resp, responses);
         } catch (IOException e) {
-            LOGGER.warn("error handling request", e);
+            LOGGER.warn("Error handling request", e);
             resp.setStatus(SC_BAD_REQUEST);
```

Comment:
This 400 response it true if the IO exception was generated  during JSON parsing.

I am not that sure that it is always the case.

(Ok not that much related to your PR...)